src/ComplexScalars/UriTrait.php

```diff
@@ -72,4 +72,83 @@ public function toNative()
     {
         return $this->uri->__toString();
     }
+
+    /**
+     * @return UriInterface
+     */
+    public function getPsr7Uri()
+    {
+        return $this->uri;
+    }
+
+    /**
+     * @param string $scheme
+     *
+     * @return UriTrait
+     */
+    public function withScheme(string $scheme)
```

Comment:
I think the return type here will break when the trait is used. Because the eventual type using the trait will not be a "UriTrait". The best thing to do is use "static" on the doc block. But unfortunately you can't use a PHP return type.

Reply:
yep, sorted